src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala

```diff
@@ -36,62 +37,75 @@ class UITestSuite extends BaseGitRepositoryBackedIntegrationTestSuite(SETUP_WITH
-      // 7 branch rows + 11 commit rows
-      Assert.assertEquals(18, branchAndCommitRowsCount)
-
-      // Let's slide out a root branch now
-      project.contextMenu.openContextMenu("develop")
-      project.contextMenu.slideOut()
-      branchAndCommitRowsCount = project.refreshModelAndGetRowCount()
-      // 5 branch rows (`develop` is no longer there) + 7 commit rows
-      // (1 commit of `allow-ownership-link` and 3 commits of `call-ws` are all gone)
-      Assert.assertEquals(13, branchAndCommitRowsCount)
-
-      project.checkoutBranch("master")
-      project.contextMenu.openContextMenu("call-ws")
-      project.contextMenu.slideOut()
-      branchAndCommitRowsCount = project.refreshModelAndGetRowCount()
-      // 4 branch rows (`call-ws` is also no longer there) + 8 commit rows
-      Assert.assertEquals(12, branchAndCommitRowsCount)
-    } catch {
-      case e: Exception =>
-        saveThreadDumpAndScreenshot()
-        throw e
+  val _saveThreadDumpWhenTestFailed = new TestWatcher() {
+    override protected def failed(e: Throwable, description: Description): Unit = {
+      val pid = probe.pid()
+      probe.screenshot("exception")
+      val threadStackTrace: String = Process("jstack " + pid) !!
+      val artifactDirectory =
+        System.getProperty("user.home") + "/artifacts/uiTest" + intelliJVersion.build + "/thread-dumps"
+      Files.createDirectories(Paths.get(artifactDirectory))
+      val file: File = new File(artifactDirectory + "/thread_dump_" + pid + ".txt")
+      val pw = new PrintWriter(file)
+      pw.write(threadStackTrace)
+      pw.close()
     }
   }
 
+  @Rule
+  def saveThreadDumpWhenTestFailed: TestWatcher = _saveThreadDumpWhenTestFailed
+
+  @Test def skipNonExistentBranches_toggleListingCommits_slideOutRoot(): Unit = {
+    project.openGitMacheteTab()
+    overwriteMacheteFile(
+      """develop
+        |  non-existent
+        |    allow-ownership-link
+        |      update-icons
+        |      build-chain
+        |    non-existent-leaf
+        |  call-ws
+        |non-existent-root
+        |master
+        |  hotfix/add-trigger""".stripMargin
+    )
+    val managedBranches = project.refreshModelAndGetManagedBranches()
+    // Non-existent branches should be skipped while causing no error (only a low-severity notification).
+    Assert.assertEquals(
+      Seq(
+        "allow-ownership-link",
+        "build-chain",
+        "call-ws",
+        "develop",
+        "hotfix/add-trigger",
+        "master",
+        "update-icons"
+      ),
+      managedBranches.toSeq.sorted
+    )
+    project.toolbar.toggleListingCommits()
+    var branchAndCommitRowsCount = project.refreshModelAndGetRowCount()
+    // 7 branch rows + 11 commit rows
+    Assert.assertEquals(18, branchAndCommitRowsCount)
+
+    // Let's slide out a root branch now
+    project.contextMenu.openContextMenu("develop")
+    project.contextMenu.slideOut()
+    project.acceptBranchDeletionOnSlideOut()
```

Comment:
Huh wait... that's not the same code as used to be inside `try ... catch` block... is it? 🤔 

That might be why the UI test is failing

Reply:
now in #skipNonExistentBranches_toggleListingCommits_slideOutRoot is the same content as on `develop`